### PR TITLE
update guide-java-spring-boot

### PR DESCRIPTION
**Pull request by Weaxs**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Fix https://github.com/pingcap/docs-cn/issues/15778

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-sample-application-java-spring-boot.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ aliases: ['/zh/tidb/dev/dev-guide-sample-application-spring-boot', '/zh/tidb/dev
 
 # 使用 Spring Boot 连接到 TiDB
 
-TiDB 是一个兼容 MySQL 的数据库。[Spring](https://spring.io/) 是当前比较流行的开源 Java 容器框架，本文选择 [Spring Boot](https://spring.io/projects/spring-boot) 作为使用 Spring 的方式。
+TiDB 是一个兼容 MySQL 的数据库。[Spring](https://spring.io/) 是当前比较流行的开源 Java 容器框架，本文选择 [Spring Boot 3](https://spring.io/projects/spring-boot) 作为使用 Spring 的方式。
```
Comment (Icemap) [inline develop/dev-guide-sample-application-java-spring-boot.md:9]: Spring Boot 3 并非 Spring Boot 名称。

File: develop/dev-guide-sample-application-java-spring-boot.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ TiDB 是一个兼容 MySQL 的数据库。[Spring](https://spring.io/) 是当前
 
 ## 前置需求
 
-- 推荐 **Java Development Kit** (JDK) **17** 及以上版本。你可以根据公司及个人需求，自行选择 [OpenJDK](https://openjdk.org/) 或 [Oracle JDK](https://www.oracle.com/hk/java/technologies/downloads/)。
+- **Java Development Kit** (JDK) **17** 及以上版本。你可以根据公司及个人需求，自行选择 [OpenJDK](https://openjdk.org/) 或 [Oracle JDK](https://www.oracle.com/hk/java/technologies/downloads/)。
```
Comment (Icemap) [inline develop/dev-guide-sample-application-java-spring-boot.md:23]: 为何要删除 `推荐` 一词？

File: develop/dev-guide-sample-application-java-spring-boot.md [hunk 3]
```diff
@@ -200,12 +200,20 @@ spring:
...
+
+```java
+@Entity
+@Table(name = "player_jpa")
+public class PlayerBean {
+}
+```
 
-Spring Data JPA 通过 `@Repository` 接口来管理数据。你需要继承 `JpaRepository` 接口，以使用其提供的增删改查函数。
+通过继承 `JpaRepository<T, ID>` 接口，注册用于管理实体的Bean；`JpaRepository<T, ID>` 接口也提供了基础的增删改查函数。
```
Comment (Icemap) [inline develop/dev-guide-sample-application-java-spring-boot.md:214]: 1. `JpaRepository` 为泛型接口的接口名称，`<T, ID>` 为泛型类型。
2. 并非接口提供增删改查函数，此处是由 Spring IoC 后提供实现类的自动装配。换言之，是由实现类提供的基础增删改查。
3. `注册用于管理实体的Bean`，此句将产生模糊指代。实体、Bean 均可指代 `PlayerBean`，且 Bean 请替换措辞。

File: develop/dev-guide-sample-application-java-spring-boot.md [hunk 4]
```diff
@@ -200,12 +200,20 @@ spring:
...
 
 ```java
-@Repository
```
Comment (Icemap) [inline develop/dev-guide-sample-application-java-spring-boot.md:208]: 此处为何删除 `@Repository`?

File: develop/dev-guide-sample-application-java-spring-boot.md (hunk 2 above)
Comment (Weaxs) [reply develop/dev-guide-sample-application-java-spring-boot.md:23]: Spring Boot 3应该最少是17+，感觉写推荐有点可以使用JDK11 or JDK8 的感觉...

File: develop/dev-guide-sample-application-java-spring-boot.md (hunk 4 above)
Comment (Weaxs) [reply develop/dev-guide-sample-application-java-spring-boot.md:208]: 这里JPA的接口可以不加`@Repository`，`JpaRepositoryFactoryBean`会给继承了`JpaRepository`的repository接口注册对应的Bean

File: develop/dev-guide-sample-application-java-spring-boot.md (hunk 3 above)
Comment (Weaxs) [reply develop/dev-guide-sample-application-java-spring-boot.md:214]: 写成这样可以吗？

> `PlayerRepository` 通过继承 `JpaRepository` 接口， 由 `JpaRepositoryFactoryBean` 为其自动注册对应的 Repository Bean。同时，`JpaRepository` 接口的默认实现类 `SimpleJpaRepository` 提供了增删改查函数的具体实现。

File: develop/dev-guide-sample-application-java-spring-boot.md (hunk 1 above)
Comment (Weaxs) [reply develop/dev-guide-sample-application-java-spring-boot.md:9]: 哦哦  主要想强调一下 spring boot 版本...我改下hhh

File: develop/dev-guide-sample-application-java-spring-boot.md (hunk 2 above)
Comment (shawn0915) [reply develop/dev-guide-sample-application-java-spring-boot.md:23]: 建议保留“推荐”，版本号也没问题。

https://spring.io/quickstart/
>A Java™ Development Kit (JDK)
>We recommend [BellSoft Liberica JDK](https://bell-sw.com/pages/downloads/#/java-17-lts) version 17.

File: develop/dev-guide-sample-application-java-spring-boot.md (hunk 4 above)
Comment (Icemap) [reply develop/dev-guide-sample-application-java-spring-boot.md:208]: 这里是因为默认的 `CrudRepository` 的实现 `SimpleJpaRepository` 存在 `@Repository` 注解吧。

FYI: https://github.com/spring-projects/spring-data-jpa/blob/main/spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/support/SimpleJpaRepository.java#L95

我这里留存 `@Repository` 有两个原因，其一是显式地声明此处是一个 Repository 实体，其二是在使用非默认实现时提供兼容。

如果您有不同看法，或者我的回答有误，烦请指正哈，感谢。

Comment (Weaxs) [reply develop/dev-guide-sample-application-java-spring-boot.md:208]: 嗯...是的，那还是加着吧，我改下hhh

File: develop/dev-guide-sample-application-java-spring-boot.md (hunk 3 above)
Comment (Icemap) [reply develop/dev-guide-sample-application-java-spring-boot.md]: LGTM

**Review by Icemap — APPROVED** [review]
LGTM

**Comment by Icemap** [comment]
感谢 @Weaxs 对 TiDB 文档的贡献。

File: develop/dev-guide-sample-application-java-spring-boot.md [hunk 5]
```diff
@@ -200,9 +200,18 @@ spring:
...
+
+```java
+@Entity
+@Table(name = "player_jpa")
+public class PlayerBean {
+}
+```
 
-Spring Data JPA 通过 `@Repository` 接口来管理数据。你需要继承 `JpaRepository` 接口，以使用其提供的增删改查函数。
+`PlayerRepository` 通过继承 `JpaRepository` 接口， 由 `JpaRepositoryFactoryBean` 为其自动注册对应的 Repository Bean。同时，`JpaRepository` 接口的默认实现类 `SimpleJpaRepository` 提供了增删改查函数的具体实现。
```
Comment (qiancai) [inline develop/dev-guide-sample-application-java-spring-boot.md:214]:
```suggestion
`PlayerRepository` 通过继承 `JpaRepository` 接口，由 `JpaRepositoryFactoryBean` 为其自动注册对应的 Repository Bean。同时，`JpaRepository` 接口的默认实现类 `SimpleJpaRepository` 提供了增删改查函数的具体实现。
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**